UnitTests/OIDAuthorizationRequestTests.m

```diff
@@ -223,6 +223,29 @@ - (void)testScopeInitializerWithManyScopesAndNoClientSecret {
                         kTestAdditionalParameterValue, @"");
 }
 
+
+/*! @brief Tests the initializer which takes a nonce
+ */
+- (void)testNonceInitializer {
+  OIDServiceConfiguration *configuration = [OIDServiceConfigurationTests testInstance];
+  OIDAuthorizationRequest *request =
+      [[OIDAuthorizationRequest alloc] initWithConfiguration:configuration
+                                                    clientId:kTestClientID
+                                                      scopes:@[]
+                                                 redirectURL:[NSURL URLWithString:kTestRedirectURL]
+                                                responseType:OIDResponseTypeCode
+                                                       nonce:kTestNonce
+                                        additionalParameters:nil];
+
+  XCTAssertEqualObjects(request.nonce, kTestNonce, @"");
```

Comment:
I recognize that you're following the pre-existing style in this file by adding an empty description of the failure in each of these asserts, but that is an odd practice IMHO. Let's remove these as they don't add anything and consider this as a first step in migrating away from that practice.

Reply:
I updated the failure message, this is what it looks like now:

<img width="1252" alt="Screenshot 2023-10-04 at 21 03 26" src="https://github.com/openid/AppAuth-iOS/assets/1566403/3d106782-dbf6-4591-9778-49bf8c802565">


Please let me know if this is good, or if I should add that to the other assertions as well.

Thank you!

Reply:
You can either add meaningful descriptions (like you have in your screenshot) or remove the parameter completely. Up to you!

Reply:
I'm sorry -- it seems like we are having a miscommunication. I'm saying to remove the argument you supply to the third parameter completely. This,

`XCTAssertEqualObjects(request.nonce, kTestNonce, @"");`

Should become

`XCTAssertEqualObjects(request.nonce, kTestNonce);`

The same very every assert in this test that you're adding. As I mentioned before, I recognize that the style is different in the other tests that were already in place in this file, but let's consider your new test as a step in the new direction.

Reply:
removed the 3rd param @mdmathias 👍